### PR TITLE
Add ListUserRevisions API

### PR DESCRIPTION
**Pull request by jtoohill**
This will eventually replace the ListEntryHistory API (#1101)

### CODE REVIEW

File: core/api/v1/keytransparency.proto [hunk 1]
```diff
@@ -372,6 +414,15 @@ service KeyTransparency {
       get: "/v1/directories/{directory_id}/users/{user_id}/history"
     };
   }
+  // ListUserRevisions returns a list of historic leaf values for a user.
+  //
+  // Clients verify their account history by observing correct values for their
+  // account over time.
+  rpc ListUserRevisions(ListUserRevisionsRequest) returns (ListUserRevisionsResponse) {
+    option (google.api.http) = {
+      get: "/v1/directories/{directory_id}/users/{user_id}/revisions"
```
Comment (therealdrake) [inline core/api/v1/keytransparency.proto]: needs to be a post I think

File: core/keyserver/keyserver.go [hunk 2]
```diff
@@ -273,6 +273,12 @@ func (s *Server) ListEntryHistory(ctx context.Context, in *pb.ListEntryHistoryRe
 	}, nil
 }
 
+// ListUserRevisions returns a list of revisions covering a period of time.
+func (s *Server) ListUserRevisions(ctx context.Context, in *pb.ListUserRevisionsRequest) (
```
Comment (therealdrake) [inline core/keyserver/keyserver.go]: the multi-lining here seems weird. Could you move the opening parens down to the line below?

File: core/api/v1/keytransparency.proto (hunk 1 above)
Comment (jtoohill) [reply core/api/v1/keytransparency.proto]: Done.

File: core/keyserver/keyserver.go (hunk 2 above)
Comment (jtoohill) [reply core/keyserver/keyserver.go]: If you don't leave the hanging open parens, Go thinks you forgot the `{` at the start of the function definition (doesn't compile).

**Pull request merged**